From 3701a8edbce7ba780815e192ffce240b1c5424e1 Mon Sep 17 00:00:00 2001
From: Mandy Lavi <mandy.lavi@freescale.com>
Date: Tue, 21 Jul 2015 14:14:14 +0300
Subject: [PATCH 318/451] dpaa_eth: read ts value according to its endianness
 type

Signed-off-by: Mandy Lavi <mandy.lavi@freescale.com>
Change-Id: I01f885821e508268b430ec9728026182abc92c4a
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../net/ethernet/freescale/dpa/dpaa_eth_common.c   |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index f11015d..228bc50 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -341,6 +341,8 @@ u64 dpa_get_timestamp_ns(const struct dpa_priv_s *priv, enum port_type rx_tx,
 	if (!ts || *ts == 0)
 		return 0;
 
+	be64_to_cpus(ts);
+
 	/* multiple DPA_PTP_NOMINAL_FREQ_PERIOD_NS for case of non power of 2 */
 	ns = *ts << DPA_PTP_NOMINAL_FREQ_PERIOD_SHIFT;
 
-- 
1.7.5.4

